### ISSUE TITLE
[BUG] При выключении "почти" всех диагностик получаю деградацию производительности анализа

### ISSUE THREAD

**Issue by qtLex** [1c-syntax/bsl-language-server#997]
**Версия**
bsl-language-server-0.14.0-RC3

**Описание ошибки диагностики**
Если выключит "почти" все диагностики, но оставить TypoDiagnostic. Получаю деградацию производительности.

**Дополнительная информация**
Пример конфиг файла [bsl-language-server.txt](https://github.com/1c-syntax/bsl-language-server/files/4337078/bsl-language-server.txt)
Если в файл добавит "Typo": false. Анализ выполнятеся на приемлемое время.

Профайлер запущенный в этот момент выдает собщение о "Potential deadlock: frozen threads found"
[potential-deadlock.txt](https://github.com/1c-syntax/bsl-language-server/files/4337079/potential-deadlock.txt)

**Comment by nixel2007** [comment]
Метод check внутри диагностики typo лежит в synchronized блоке. Сам метод долгий и то, что там появляется ожидание - это нормально. Странно, что треды ждут по 10 секунд, но без исходников конфигурации такое отловить будет сложно.

**Comment by qtLex** [comment]
Проверю зависит ли это от конфигурации.

Ожидание, это нормально. Но когда время с 7 минут просело до ~7 часов, тут явно проблемы

**Comment by nixel2007** [comment]
Да, это конечно же проблема :)

**Comment by qtLex** [comment]
Попробовал на нескольких конфигурациях. Нет явной зависимости от конфигурации.
Распространенных воспроизводится на УХ 3.0.10.13.

**Comment by EightM** [comment]
xMax указываешь сразу при запуске?

**Comment by qtLex** [comment]
Вообще не указываю.

**Comment by EightM** [comment]
Попробуй

**Comment by nixel2007** [comment]
@qtLex можешь проверить на вот этой сборке?

https://github.com/1c-syntax/bsl-language-server/pull/998

**Comment by qtLex** [comment]
> @qtLex можешь проверить на вот этой сборке?
> 
> #998

@nixel2007 Не помогло.

**Comment by asosnoviy** [comment]
@qtLex релиз вышел. Перфоманс починили?

**Comment by qtLex** [comment]
@asosnoviy Нет. все по прежнему. Без конфига с выключенными диагностиками 10 мин, с конфигом за час 39%.

**Comment by theshadowco** [comment]
@qtLex проблема атуальна?

**Comment by qtLex** [comment]
Проблема все еще есть, но у меня не болит. Это очень редкий сценарий использования. Думаю можно закрыть.

**Comment by MinimaJack** [comment]
@qtLex возможно проблема в железе? Swap? На другом пробовал?

**Comment by qtLex** [comment]
>@qtLex возможно проблема в железе? Swap? На другом пробовал?

На трех разных компах и двух конфигурациях. Реакция одна

**Comment by otymko** [comment]
@qtLex проверил сейчас на комете "6a75b0c3d5ceeaf15bf7973fe3176303f8d3deee", проблема твоя не воспроизвелась. Еще актуально?
